### DISCUSSION TITLE
remove ui-message-info when file chooser is opened

### DISCUSSION THREAD

**Discussion by Alirzrezaei** [primefaces/community#1303] [PrimeFaces] [answered]
I am using primefaces v13 and the p:fileUpload tag to upload a file. After a successful or failed upload, a message is shown on the screen using from Java bean class. 
```

FacesMessage message = new FacesMessage(FacesMessage.SEVERITY_INFO, "my msg title",
					"my message.");
			FacesContext.getCurrentInstance().addMessage(null, message);
```

I want to remove the message when the user clicks again to open the file chooser. I found no attribute related to p:fileUpload to do so. I also tried the following code, but it was not working. 
```
`<script>
    // Add this script to handle the 'Select File' button click
    $(document).ready(function () {
        // Assuming the 'Select File' button has a class named 'ui-fileupload-choose'
        $('.ui-fileupload-choose').on('click', function () {
            // Clear the messages when the button is clicked
            clearMessages();
        });
    });

    function clearMessages() {
        // Find and remove all messages from the page
        $('.ui-messages').children().remove();
    }
</script>`
```

**Comment by melloware** [comment]
@Alirzrezaei it works fine. 

1. Go to the Showcase: https://www.primefaces.org/showcase/ui/message/messages.xhtml?jfwid=bcf30
2. Click the "Info" button.
3. Open F12 Console and type `$('.ui-messages').children().remove();`

The message is removed from the screen.

**Reply by Alirzrezaei** [reply]
@melloware thanks for your guide. It runs only one time, I want it to remove the messages every time a user clicks on Select File and the dialog box is opened. 

![fileUpload](https://github.com/primefaces/community/assets/32749480/cddabd52-9aa4-4a3d-b46f-bad93ef2fcf6)

**Reply by Alirzrezaei** [reply]
@melloware As I search there is no onclick event for fileUpload. I thought to use onstart, but it is at the beginning of file upload and I need to remove the message before.

**Reply by melloware** [reply, marked as answer]
So I just did this and it works for me in the showcase?

```js
$('.ui-fileupload-choose').on('click', () => $('.ui-messages').children().remove());
```
